src/components/button/CircleButton.vue

```diff
@@ -1,17 +1,27 @@
 <template>
-    <el-row>
-        <el-button icon="el-icon-search" circle></el-button>
-        <el-button type="primary" icon="el-icon-edit" circle></el-button>
-        <el-button type="success" icon="el-icon-check" circle></el-button>
-        <el-button type="info" icon="el-icon-message" circle></el-button>
-        <el-button type="warning" icon="el-icon-star-off" circle></el-button>
-        <el-button type="danger" icon="el-icon-delete" circle></el-button>
-    </el-row>
+  <el-button size="medium" :type="type" :icon="icon" circle>
+    <slot></slot>
+  </el-button>
+    <!-- <el-row>
+        <el-button size="medium" icon="el-icon-search" circle></el-button>
+        <el-button size="medium" type="primary" icon="el-icon-edit" circle></el-button>
+        <el-button size="medium" type="success" icon="el-icon-check" circle></el-button>
+        <el-button size="medium" type="info" icon="el-icon-message" circle></el-button>
+        <el-button size="medium" type="warning" icon="el-icon-star-off" circle></el-button>
+        <el-button size="medium" type="danger" icon="el-icon-delete" circle></el-button>
+    </el-row> -->
 </template>
 
 <script>
 export default {
-  name: 'CircleButton'
+  name: 'CircleButton',
+  props: {
+    type: {
+      type: String,
+      icon: String,
+      default: ''
+    }
```

Comment:
The structure for a prop is this:
```
nameOfProp: {
  type: variableTrpe e.g Number, String, Boolean etc
  default: The default value is optional. However, if put it should match the type
}
```
So  here should be:
```
props: {
  type: {
      type: String,
      default: ''
  },
  icon: {
      type: String,
      default: ''
  }
}
```
Further reading on props can be done here:
https://vuejs.org/v2/guide/components-props.html